hazelcast/src/main/java/com/hazelcast/spi/impl/classicscheduler/ScheduleQueue.java

```diff
@@ -0,0 +1,69 @@
+package com.hazelcast.spi.impl.classicscheduler;
+
+/**
+ * The ScheduleQueue is a kind of priority queue where 'tasks' are queued for scheduling.
+ * <p/>
+ * ScheduleQueue support concurrent producers but only need to support single consumers.
```

Comment:
why the interface specifies concurrency properties? MPSC? this is a concern of implementation, isn't it?